### PR TITLE
btrfs: correctly handle non-snapshot inodes on delete

### PR DESCRIPTION
**Pull request by stevvooe**
Signed-off-by: Stephen J Day <stephen.day@docker.com>

cc @samuelkarp @dmcgowan

### CODE REVIEW

**Comment by dmcgowan** [comment]
LGTM

File: btrfs.go [hunk 1]
```diff
@@ -328,11 +328,11 @@ func SubvolDelete(path string) error {
 		}
 
 		if err := isFileInfoSubvol(fi); err != nil {
-			return err
+			return nil
```
Comment (samuelkarp) [inline btrfs.go:331]: If you're ignoring `err`, what's the point of calling `isFileInfoSubvol` in the first place?

Comment (stevvooe) [reply btrfs.go:331]: `isFileInfoSubvol` is used in cases where it is an error to call the function on something that is not a subvol. This was just an oversight in usage. Might be better to change to `(bool, error)` or something like that.

Comment (samuelkarp) [reply btrfs.go:331]: `(bool, error)` would definitely be a better signature for this.

Comment (stevvooe) [reply btrfs.go:331]: @samuelkarp It makes it kind of a pain to use.

**Pull request merged**